mm2src/coins/utxo.rs

```diff
@@ -610,6 +619,11 @@ pub struct UtxoCoinFields {
     /// This abortable system is used to spawn coin's related futures that should be aborted on coin deactivation
     /// and on [`MmArc::stop`].
     pub abortable_system: AbortableQueue,
+    pub(crate) ctx: MmWeak,
+    /// This is used for balance event streaming implementation for UTXOs.
+    /// If balance event streaming isn't enabled, this value will always be `None`; otherwise,
+    /// it will be used for receiving scripthash notifications to re-fetch balances.
+    scripthash_notification_handler: ScripthashNotificationHandler,
```

Comment:
> I guess I will need to implement this in the future for other coins, e.g. when tendermint has multiple addresses

When HD is fully functional for all protocols in mm2, then we probably need to change some of the implementation design (like specific trait for common usecases).

Reply:
I already did a big refactor for HD wallet in the EVM HD PR to make a lot of functions more generic, I will keep you above comment in mind to see how to make this more common/generic too. I will resolve this since I added it to my HD wallet checklist.